buildflow/runtime/ray_io/gcs_io.py

```diff
@@ -0,0 +1,136 @@
+import dataclasses
+import logging
+from typing import Any, Dict, List, Optional
+
+from google.api_core import exceptions
+from google.cloud import storage
+
+from buildflow.api import io
+from buildflow.runtime.ray_io import pubsub_io
+from buildflow.runtime.ray_io import pubsub_utils
+
+
+@dataclasses.dataclass
+class GCSFileEvent:
+    # Metadata about that action taken.
+    metadata: Dict[str, Any]
+
+    @property
+    def blob(self) -> bytes:
+        if self.metadata['eventType'] == 'OBJECT_DELETE':
+            raise ValueError('Can\'t fetch blob for `OBJECT_DELETE` event.')
+        client = storage.Client()
+        bucket = client.bucket(bucket_name=self.metadata['bucketId'])
+        blob = bucket.get_blob(self.metadata['objectId'])
+        return blob.download_as_bytes()
```

Comment:
nit: What if you just return the blob object?

Reply:
The metadata field covers most of it so I thought it would be better just to get the bytes.